framework/include/parser/Parser.h

```diff
@@ -243,6 +243,9 @@ class Parser : public ConsoleStreamInterface, public hit::Walker
   /// The current stream object used for capturing errors during extraction
   std::ostringstream * _current_error_stream;
 
+  /// Tracks whether a deprecated param has had its warning message printed already.
+  std::unordered_set<std::string> _deprec_param_tracker;
```

Comment:
@permcody Here are the C++ changes at least -- Not 100% sure this is the best choice of container here or if there's a way to make this more elegant in general

Reply:
I figured unordered_set because it doesn't need to be sorted, just iterated, and I believe average case complexity is faster than vector?

Reply:
Your first priority is to get something that works and this does that. If it's in a time-sensitive area, then you can take a deeper look at optimizing something like container types. For what it's worth, vectors and arrays are by far the fastest containers for pretty much all but the most diabolical cases (like repeatedly inserting or removing in the middle). `unordered_set`s are surprisingly fast, almost always faster than just plain `set` even in cases where you'd expect `set`s underlying data structure to really fly.